spec/order_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'order'
+
+describe Order do
+  let(:pasta) { double :pasta }
+  let(:paneer_tikka_masala) { double :paneer_tikka_masala}
+  let(:pizza) { double :pizza}
+  let(:dish) { double(:dish, name: paneer_tikka_masala, price: 10) }
+  let(:new_dish) { double(:new_dish, name: pizza, price: 15) }
+  let(:menu) { double(:menu, menu: [dish, new_dish])}
+  
+  it 'raises error if you attempt to delete an item from the basket, which is not in the basket' do
+    expect {subject.remove_dish_from_basket(paneer_tikka_masala) }.to raise_error "This dish is not currently in your basket"
+  end
+
+  it 'outputs "Your basket is empty if you attempt to calculate the cost of an empty basket' do
+    expect { subject.calculate_order_price }.to output("Your basket is empty\n").to_stdout
+  end
+
+  context 'with mocking menu class' do
+
+    before do
+      allow(Menu).to receive(:new).and_return(menu)
```

Comment:
So you can avoid stubbing `Menu.new` if you are able dependency-inject the `menu` double into `Order.new` (like mentioned by my comment above), rather than having `Order#initialize` controlling how it creates a new Menu